gst/nnstreamer/nnstreamer.c

```diff
@@ -48,48 +48,55 @@
 #include <gst/gst.h>
 #include <gst/gstplugin.h>
 
-#define NNSTREAMER_PLUGIN(name) \
-  extern gboolean G_PASTE(nnstreamer_export_, name) (GstPlugin *plugin)
-
-NNSTREAMER_PLUGIN (tensor_converter);
-NNSTREAMER_PLUGIN (tensor_aggregator);
-NNSTREAMER_PLUGIN (tensor_decoder);
-NNSTREAMER_PLUGIN (tensor_demux);
-NNSTREAMER_PLUGIN (tensor_merge);
-NNSTREAMER_PLUGIN (tensor_mux);
-NNSTREAMER_PLUGIN (tensor_sink);
-NNSTREAMER_PLUGIN (tensor_src_iio);
-NNSTREAMER_PLUGIN (tensor_split);
-NNSTREAMER_PLUGIN (tensor_transform);
-NNSTREAMER_PLUGIN (tensor_filter);
-NNSTREAMER_PLUGIN (tensor_reposink);
-NNSTREAMER_PLUGIN (tensor_reposrc);
-
-#define NNSTREAMER_INIT(name, plugin) \
-  do { \
-    if (!G_PASTE(nnstreamer_export_, name)(plugin)) \
-      return FALSE; \
-  } while (0);
+#include "tensor_aggregator/tensor_aggregator.h"
+#include "tensor_converter/tensor_converter.h"
+#include "tensor_decoder/tensordec.h"
+#include "tensor_demux/gsttensordemux.h"
+#include "tensor_filter/tensor_filter.h"
+#include "tensor_merge/gsttensormerge.h"
+#include "tensor_mux/gsttensormux.h"
+#include "tensor_reposink/tensor_reposink.h"
+#include "tensor_reposrc/tensor_reposrc.h"
+#include "tensor_sink/tensor_sink.h"
+#include "tensor_source/tensor_src_iio.h"
+#include "tensor_split/gsttensorsplit.h"
+#include "tensor_transform/tensor_transform.h"
 
 /**
  * @brief Function to initialize all nnstreamer elements
  */
 static gboolean
 gst_nnstreamer_init (GstPlugin * plugin)
 {
-  NNSTREAMER_INIT (tensor_converter, plugin);
-  NNSTREAMER_INIT (tensor_aggregator, plugin);
-  NNSTREAMER_INIT (tensor_decoder, plugin);
-  NNSTREAMER_INIT (tensor_demux, plugin);
-  NNSTREAMER_INIT (tensor_merge, plugin);
-  NNSTREAMER_INIT (tensor_mux, plugin);
-  NNSTREAMER_INIT (tensor_sink, plugin);
-  NNSTREAMER_INIT (tensor_src_iio, plugin);
-  NNSTREAMER_INIT (tensor_split, plugin);
-  NNSTREAMER_INIT (tensor_transform, plugin);
-  NNSTREAMER_INIT (tensor_filter, plugin);
-  NNSTREAMER_INIT (tensor_reposink, plugin);
-  NNSTREAMER_INIT (tensor_reposrc, plugin);
+  if (!gst_element_register (plugin, "tensor_aggregator", GST_RANK_NONE,
+          GST_TYPE_TENSOR_AGGREGATOR) ||
+      !gst_element_register (plugin, "tensor_converter", GST_RANK_NONE,
```

Comment:
How about using macro to make code more brief?

Reply:
Now updated to add macro for registration.